### PR TITLE
fix: return non-zero if there were errors in processing

### LINKED ISSUE #80: Github action still seems successful even if collaborators weren't added
### Bug description

If there are errors adding any users, such as a typo in the repo name, the github action still succeeds.  The error shows up with a helpful red X in the log message, but it's not obvious that the logs need to be checked since the job was successful.

In my case, the issue was that I had run out of seats in our paid plan, which also wasn't easy to figure out because the error message was just "problem detected".

### Steps to reproduce

1. Create a new file under `repos` called `dummy.yml` (assuming there isn't a repo called `dummy`!)
```
dummy:
  some-group:
    type: "group"
    permissions: "read"
```
2. Run the "Update Outside Collaborators" job
3. Job will complete successfully, even though there are errors in the log

### Expected behavior

I'd expect the job would fail and the logs would have enough details to diagnose

### Example repository

_No response_

### Additional context

I have a simple fix that I can contribute back, but I'm not a Ruby developer, so it might not be idiomatic.

### PR DESCRIPTION
**Pull request by renaissanceGeek**
Fixes #80 very simply.  There may be something more elegant, but at least this helped my use case!

### CODE REVIEW

File: scripts/outside-collaborators-handler.rb [hunk 1]
```diff
@@ -114,8 +114,8 @@ def add_repo_collaborator(repo, user, auth)
             check_and_wait_until_reset
             begin
                 $client.add_collaborator(repo, user, permission: auth__)
-            rescue
-                puts " - problem detected ❌"
+            rescue StandardError => e
+                puts " - problem detected #{e.inspect}❌"
```
Comment (pattacini) [inline scripts/outside-collaborators-handler.rb:118]:
```suggestion
                puts " - problem detected: #{e.inspect}❌"
```

**Review by pattacini — CHANGES_REQUESTED** [review]
I like the simple logic @renaissanceGeek 👍🏻 
I'm just wondering if you're willing to deal also with the error conditions caught inside `add_repo_collaborator` that now won't make the job exit 1.

You may turn `has_errors` into a global var (e.g., `$has_errors`) as `$org`  to handle those 3 further conditions.

Instead, a better way would be to make `add_repo_collaborator` return false when one of those 3 exceptions occurs so that you'll be able to account for that with `has_errors` locally to the main.

### Just a note for myself
Differently from this logic, `check-automated-repositories` shall exit as soon as an exception gets found as it is done right now so that the PR check can provide immediate feedback.

**Comment by renaissanceGeek** [comment]
OK, I think I resolved your requests.  I tried testing various error conditions that could come up in add_repo-collaborator and successfully failed ;-)

**Review by pattacini — APPROVED** [review]
Fantastic job @renaissanceGeek 🚀 
I've only added a missing `return true` and dealt with just another call to the function.

Thanks again for your contribution!
Appreciated.

**Pull request merged**